packages/design-system-react/src/components/text/Text.types.ts

```diff
@@ -60,6 +60,10 @@ export type TextProps = {
    * @default TextColor.TextDefault
    */
   color?: TextColor;
+  /**
+   * Optional prop to pass to the underlying HTML element
+   */
+  'data-testid'?: string;
```

Comment:
Need to further investigate why `data-` attributes are not accepted when using prop objects.  

Applying a `data-testid` attribute directly to the `Text` component works:  
![Screenshot 2025-01-30 at 3 59 22 PM](https://github.com/user-attachments/assets/6f09dd9c-6cc9-4748-83bb-ee7400dab590)  

However, applying it through a prop object does not:  
![Screenshot 2025-01-30 at 3 58 42 PM](https://github.com/user-attachments/assets/e2d301ae-38cd-424e-83f8-dbafdcb49724)

Reply:
Oh duh, I wasn't spreading props on the Text object 🤦